### PR TITLE
Mobile create pin

### PR DESCRIPTION
**Pull request by cgjohn**
https://github.com/user-attachments/assets/41036932-d0fe-4e7b-9fe4-08fa13970a7f

implements [this flow from figma](https://www.figma.com/design/ejI1OJnIyNbqiPw94FwzDp/Mobile-App?node-id=3953-10150&t=XhIPPuzuRq15FJGo-4)

### CODE REVIEW

**Review by dguenther — APPROVED** [review]
LGTM if it's functional! Might want to grab a review from Dan as well though since he's probably more informed on the frontend 😄

File: packages/tackle-box/lib/components/PinInput/PinInput.tsx
Comment (dgca) [inline packages/tackle-box/lib/components/PinInput/PinInput.tsx:150]: 👀

**Review by dgca — APPROVED** [review]
Probably wanna remove that console.log, but lgtm otherwise

**Pull request merged**
